client/state/login/reducer.js

```diff
@@ -67,7 +67,9 @@ export const rememberMe = createReducer( null, {
 export const isFormDisabled = createReducer( null, {
 	[ LOGIN_REQUEST ]: () => true,
 	[ LOGIN_REQUEST_FAILURE ]: () => false,
-	[ LOGIN_REQUEST_SUCCESS ]: () => true,
+	// Disable the login form when there is no 2FA data because the page is going to reload.
+	// Keep it enabled for 2FA because user might come back to the login screen.
+	[ LOGIN_REQUEST_SUCCESS ]: ( state, action ) => ! get( action, 'data.user_id', false ),
```

Comment:
What about using resetting `isFormDisabled` when `ROUTE_SET` is triggered instead?